### PR TITLE
Update "What is IaC" page

### PR DESCRIPTION
**Pull request by scottslowe**
## Description

This PR updates the "What is Infrastructure as Code" page on the Pulumi web site.

## Checklist:

- [x] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [x] I have manually confirmed that all new links work.

### CODE REVIEW

File: themes/default/content/what-is/what-is-infrastructure-as-code.md [hunk 1]
```diff
@@ -62,85 +62,105 @@ video:
...
+## Why is Infrastructure as Code Important?
+
+Infrastructure as code is important because of three significant trends, all of them happening at the same time.
+
+### The Transition to the Cloud
 
 One trend, of course, is the ongoing transition to the cloud. More and more companies are shifting workloads from on-premises infrastructure to cloud environments. Cloud-based infrastructure is provisioned via APIs, and, as a result, can be easily managed with infrastructure as code tools.
 
-### Cloud modernization
+### Cloud Modernization
```
Comment (aaronkao) [inline themes/default/content/what-is/what-is-infrastructure-as-code.md:86]: is transition to cloud and cloud modernization sorta redundant with each other? i wonder if the other trend should be more process or people oriented

File: themes/default/content/what-is/what-is-infrastructure-as-code.md [hunk 2]
```diff
@@ -62,85 +62,105 @@ video:
...
+
+The second trend is cloud modernization. After organizations migrate to the cloud, they tend to look for opportunities to maximize the value they get from their cloud environment. This frequently involves adopting technologies such as serverless, containers and Kubernetes. When applied correctly, these technologies enable teams to deliver value more quickly. In some cases, cloud modernization involves using managed services to offload some of the heavy lifting to the cloud provider. These technologies and services generally require a more granular management of infrastructure. Stitching together all the primitives that the cloud provider offers into solutions that serve the business is a great fit for infrastructure as code.
```
Comment (aaronkao) [inline themes/default/content/what-is/what-is-infrastructure-as-code.md:88]: should we call out the ephemerality and statelessness of some of these compute primitives?

File: themes/default/content/what-is/what-is-infrastructure-as-code.md [hunk 3]
```diff
@@ -62,85 +62,105 @@ video:
...
+## What are the Key Elements of Infrastructure as Code?
 
-For teams managing tens or hundreds of cloud resources that change once every few months, it may still be possible to manage infrastructure with point-and-click or scripts. But the more common situation today is that teams are managing thousands or tens of thousands of resources that change daily or even hourly. Infrastructure as code is how you take control of that kind of complexity.
+The key elements of infrastructure as code are the same key elements you'd find in the majority of software engineering environments. These include:
+
+1. **An infrastructure as code tool:** Infrastructure as code tools may be provided by and limited to a single cloud provider (AWS Cloudformation is one example), or may support multiple cloud providers. Tools may be limited to supporting YAML or JSON; may require the use of a specialized and proprietary domain-specific language (DSL); or may support the use of general purpose programming languages such as TypeScript/JavaScript, C#, Go, Python, and Java.
```
Comment (aaronkao) [inline themes/default/content/what-is/what-is-infrastructure-as-code.md:100]: its a bit tautological to define elements of IaC with IaC tool right?

Should this instead be like programming model? or resource model?

**Review by aaronkao — COMMENTED** [review]
Some comments to think about

**Comment by aaronkao** [comment]
Eli mentioned to incorporate some of the elements from Joe's IaC posts into this page

https://thenewstack.io/author/joe-duffy/. The first post has some elements about the benefits and the history that could be good to pull in. The Benefits of Code pieces is very opinionated but still stays a bit neutral which could be good.

File: themes/default/content/what-is/what-is-infrastructure-as-code.md (hunk 1 above)
Comment (scottslowe) [reply themes/default/content/what-is/what-is-infrastructure-as-code.md:86]: For the short-term future, I'm of the opinion that "transitioning to the cloud" and "cloud modernization" (or perhaps we could use "cloud optimization") are different and distinct. The first captures those companies that are largely involved in a "lift-and-shift" effort to get onto a cloud provider, whereas the second involves a greater level of cloud maturity and natively leveraging cloud services in a more effective and efficient way. Does that make sense?

File: themes/default/content/what-is/what-is-infrastructure-as-code.md (hunk 2 above)
Comment (scottslowe) [reply themes/default/content/what-is/what-is-infrastructure-as-code.md:88]: Yes, that's probably a good idea (and my response to your comment above also gives me some ideas for additional changes in this section).

File: themes/default/content/what-is/what-is-infrastructure-as-code.md (hunk 3 above)
Comment (scottslowe) [reply themes/default/content/what-is/what-is-infrastructure-as-code.md:100]: I have mixed feelings on this. On one hand, we define IaC (in the first paragraph) as "bringing software engineering principles, approaches, and tools into the cloud infrastructure space", which means that including an IaC tool is appropriate. OTOH, your comment has merit. However, I don't feel that "programming model" or "resource model" are appropriate here. In reality, you can't do IaC without an IaC tool. Perhaps some rewording would be appropriate here.

File: themes/default/content/what-is/what-is-infrastructure-as-code.md [hunk 4]
```diff
@@ -97,7 +97,7 @@ For teams managing tens or hundreds of cloud resources that change once every fe
...
 
-1. **An infrastructure as code tool:** Infrastructure as code tools may be provided by and limited to a single cloud provider (AWS Cloudformation is one example), or may support multiple cloud providers. Tools may be limited to supporting YAML or JSON; may require the use of a specialized and proprietary domain-specific language (DSL); or may support the use of general purpose programming languages such as TypeScript/JavaScript, C#, Go, Python, and Java.
+1. **An infrastructure as code tool:** It may seem a bit tautological to say you need an infrastructure as code tool to do infrastructure as code, but you do need a tool that is responsible for translating the infrastructure as code documents into something that the cloud provider APIs understand and can use. Infrastructure as code tools may be provided by and limited to a single cloud provider (AWS Cloudformation is one example), or may support multiple cloud providers. Tools may be limited to supporting YAML or JSON; may require the use of a specialized and proprietary domain-specific language (DSL); or may support the use of general purpose programming languages such as TypeScript/JavaScript, C#, Go, Python, and Java.
```
[the remark this answers is not in the document]
Comment (scottslowe) [reply themes/default/content/what-is/what-is-infrastructure-as-code.md:100]: Yeah, I can probably find a better word here! :-)

File: themes/default/content/what-is/what-is-infrastructure-as-code.md (hunk 3 above)
Comment (aaronkao) [reply themes/default/content/what-is/what-is-infrastructure-as-code.md]: maybe IaC engine makes it sound a bit more meaty?

File: themes/default/content/what-is/what-is-infrastructure-as-code.md
Comment (aaronkao) [inline themes/default/content/what-is/what-is-infrastructure-as-code.md:1]: the IaC video is just way too much in your face. I think it should be in moved to half the page in the right side of the column and the first paragraph or so to the left column of it. I want people to jump directly into reading and have the optionality of watching the video, but have the first thing be the video just seems like a waste of space

File: themes/default/content/what-is/what-is-infrastructure-as-code.md (hunk 3 above)
Comment (scottslowe) [reply themes/default/content/what-is/what-is-infrastructure-as-code.md]: I've made some changes here that I think address the concern, but your feedback is welcome.

File: themes/default/content/what-is/what-is-infrastructure-as-code.md
Comment (scottslowe) [reply themes/default/content/what-is/what-is-infrastructure-as-code.md:1]: I'm working out how to move the video to later in the article (or at the end).

**Comment by DadGPT** [comment]
Some SEO Notes:
- Mentioned elsewhere but we need the main header to be an h1, not an h2
- Video shouldn't be the first piece of content below the header... recommend we have an h2 (ideally "What is Infrastructure as Code" as the h2) and intro paragraph of copy before a YouTube embed.  Also, is this the best video we have for this specific page?
- Might be worthwhile to fit in a few more acronym mentions ("IaC")... the acronym has less search volume but still a valuable target.  Currently we don't say "What is IaC" even once in the article, so we may be leaving some longer tail visibility on the table if we don't add IaC in more.

**Comment by scottslowe** [comment]
@DadGPT Thanks for the feedback!

* Regarding the H1: this change is made on the site, but won't be reflected in this particular PR (the change to make the rainbow text an H1 was merged after this branch was created, so this branch doesn't have that change).
* I've moved the video (this was something others had recommended anyway). Unfortunately, at this time this is the best video we have.
* Can you offer any suggestions on how I can work in the acronym "IaC" more? Is it a matter of being more "inconsistent" about using the full term vs. using the acronym? In other words, sometimes use the full term and sometimes use the acronym?

File: themes/default/content/what-is/what-is-infrastructure-as-code.md
Comment (scottslowe) [reply themes/default/content/what-is/what-is-infrastructure-as-code.md:1]: Video has been moved toward the bottom of the article

**Comment by DadGPT** [comment]
Thanks @scottslowe - I like the way you put it, being more 'inconsistent' is exactly how I would approach it, especially throughout the body.  I would take the 38 instances of "infrastructure as code" and switch 5 or 6 of those references to IaC... I think that the idea that they're interchangeable makes it ok to switch back and forth, and then we get a little better keyword density.

**Pull request merged**